icu4c/source/test/intltest/msfmrgts.cpp

```diff
@@ -987,11 +988,29 @@ void MessageFormatRegressionTest::Test4112104()
     delete format;
 }
 
+void MessageFormatRegressionTest::TestICU12584() {
+    // 42 placeholders.
+    // We were able to handle only 10 due to the bug ICU-12584.
+     UnicodeString pattern("{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}" \
```

Comment:
The trailing backslash is ugly and unnecessary... and you can use the u prefix to avoid string conversion.
```
     UnicodeString pattern(
        u"{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}"
        u"{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}"
        u"{3}{1}{2}{3}");
```

Reply:
Done